src/transformers/models/bloom/modeling_bloom.py

```diff
@@ -359,12 +336,12 @@ def forward(
         output_attentions=False,
     ):
         # hidden_states: [batch_size, seq_length, hidden_size]
-        # repeat alibi tensor with the batch size
-        alibi = alibi.repeat(hidden_states.shape[0], 1, 1).to(hidden_states.device)
-
         # apply preprocessing if the input is padded
         if attention_mask is not None and 0 in attention_mask:
```

Comment:
```suggestion
        if attention_mask is not None:
```
I'd recommend not to check on 0 in attention mask: this triggers host-device synchronization, which slows down GPU and paralyzes TPU.

Reply:
Very good to know ! Thanks for the suggestion